src/ripple/RippleEngine.ts

```diff
@@ -338,18 +339,23 @@ export class XrpEngine extends CurrencyEngine<
       )
       return
     }
-    const takerDenom =
-      tokenId == null
-        ? this.currencyInfo.denominations[0]
-        : this.builtinTokens[tokenId].denominations[0]
+
+    let takerDenom
+    if (tokenId == null) {
+      takerDenom = this.currencyInfo.denominations[0]
+    } else {
+      const builtinToken = this.builtinTokens[tokenId]
+      if (builtinToken == null) return
+      takerDenom = this.builtinTokens[tokenId]?.denominations[0]
```

Comment:
change to
`takerDenom = builtInToken.denominations[0]`

This will fix typescript issue